src/utilities/hex-color-helper.ts

```diff
@@ -26,3 +28,55 @@ export function formatHexColor(color: string): HexColor {
 function isValidHex(color: string): color is HexColor {
   return /^#[0-9A-F]{6}$/i.test(color);
 }
+
+/** Returns true if provided HexColor is readable on Discord dark, light, and AMOLED modes*/
+export function validateHexColorContrast(color: HexColor): boolean {
+  const DARK_MODE_RGB: RGB = [54, 57, 63];
+  const LIGHT_MODE_RGB: RGB = [255, 255, 255];
+  const AMOLED_MODE_RGB: RGB = [0, 0, 0];
+  const MINIMUM_DARK_CONTRAST = 1.5;
+  const MINIMUM_LIGHT_CONTRAST = 1;
+
+  const rgbToTest = hexToRgb(color);
+
+  const darkModeReadable =
+    contrast(rgbToTest, DARK_MODE_RGB) > MINIMUM_DARK_CONTRAST;
+  const lightModeReadable =
+    contrast(rgbToTest, LIGHT_MODE_RGB) > MINIMUM_LIGHT_CONTRAST;
+  const amoledModeReadable =
+    contrast(rgbToTest, AMOLED_MODE_RGB) > MINIMUM_DARK_CONTRAST;
+
+  return darkModeReadable && lightModeReadable && amoledModeReadable;
+}
+
+function hexToRgb(hexColor: HexColor): RGB {
+  const hexComponents = hexColor.match(/[a-f\d]{2}/gi);
+
+  if (!hexComponents || hexComponents.length !== 3) {
+    throw new InvalidColorStringError(hexColor);
+  }
+
+  return [
+    parseInt(hexComponents[0], 16),
+    parseInt(hexComponents[1], 16),
+    parseInt(hexComponents[2], 16)
+  ];
+}
+
+function contrast(rgb1: RGB, rgb2: RGB): number {
+  var lum1 = luminance(rgb1[0], rgb1[1], rgb1[2]);
+  var lum2 = luminance(rgb2[0], rgb2[1], rgb2[2]);
+  var brightest = Math.max(lum1, lum2);
+  var darkest = Math.min(lum1, lum2);
+
+  console.log((brightest + 0.05) / (darkest + 0.05));
+  return (brightest + 0.05) / (darkest + 0.05);
+}
+
+function luminance(r: number, g: number, b: number): number {
+  var a = [r, g, b].map(function (v) {
```

Comment:
I know you copied these methods but is it possible we can swap with `let` or `const`? When I implement some of the eslint rules I know one will be disallowing `var`

Reply:
Definitely agree with now allowing var. Also, I would recommend using the () => { syntax as well.